dash/llmq/llmq.go

```diff
@@ -0,0 +1,256 @@
+package llmq
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"sort"
+
+	bls "github.com/dashpay/bls-signatures/go-bindings"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/bls12381"
+)
+
+var (
+	errThresholdInvalid      = errors.New("threshold must be greater than 0")
+	errKeySharesNotGenerated = errors.New("to initialize shares you must generate the keys")
+)
+
+type optionFunc func(c *llmqConfig)
+
+// Data contains pre generated keys/shares/signatures for the participants of LLMQ network
+type Data struct {
+	Threshold        int
+	ProTxHashes      []crypto.ProTxHash
+	PrivKeys         []crypto.PrivKey
+	PubKeys          []crypto.PubKey
```

Comment:
Maybe this can be just a func to derive pubkey from priv key, instead of storing redundant data? (I don't know performance impact)